kratos/solving_strategies/strategies/explicit_solving_strategy_bfecc.h

```diff
@@ -0,0 +1,546 @@
+     */
+    virtual void AssignSettings(const Parameters ThisParameters) override
+    {
+        BaseType::AssignSettings(ThisParameters);
+    }
+
+    /**
+     * @brief Returns the name of the class as used in the settings (snake_case format)
+     * @return The name of the class
+     */
+    static std::string Name()
+    {
+        std::stringstream s;
+        s << "explicit_solving_strategy_bfecc";
+        return s.str();
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void Initialize() override
+    {
+        BaseType::Initialize();
+        BaseType::GetModelPart().GetProcessInfo().SetValue(TIME_INTEGRATION_THETA, 0.0);
+    }
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        std::stringstream ss;
+        ss << "ExplicitSolvingStrategyBFECC";
+        return ss.str();
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream &rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream &rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    void SolveWithLumpedMassMatrix() override
+    {
+        KRATOS_TRY
+
+        // Validate time step size
+        const double dt = BaseType::GetDeltaTime();
+        KRATOS_ERROR_IF(dt < 1.0e-12) << "ProcessInfo DELTA_TIME is close to zero." << std::endl;
+
+        // Stash the prev step solution
+        const auto original_starting_values = ExtractSolutionStepData(1);
+
+        // Take step forward
+        {
+            const auto u_fixed = CopySolutionStepData(1, 0);
+            PerformSubstep(u_fixed, FORWARD);
+        }
+
+        // Overwrite previous step solution with backwards step
+        {
+            const auto u_fixed = CopySolutionStepData(0, 1);
+            PerformSubstep(u_fixed, BACKWARD);
+        }
+
+        // Correct error with the previous step solution
+        CorrectErrorAfterForwardsAndBackwards(original_starting_values);
+
+        // Final update
+        {
+            const auto u_fixed = CopySolutionStepData(1, 0);
+            PerformSubstep(u_fixed, FINAL);
+        }
+
+        KRATOS_CATCH("");
+    }
+
+    /**
+     * Ovrwrite the destination buffer position with data from the source buffer position.
+     * Additionally, we save in an auxiliary vector the value of the fixed DOFs in the destination buffer position.
+     */
+    LocalSystemVectorType CopySolutionStepData(const SizeType source, const SizeType destination)
```

Comment:
```suggestion
    LocalSystemVectorType CopySolutionStepData(
        const SizeType source,
        const SizeType destination)
```
Minor style guide suggestion.